packages/core/admin/admin/src/content-manager/history/components/VersionsList.tsx

```diff
@@ -1,19 +1,259 @@
+    addSuffix: true,
+    // @ts-expect-error - I don't know how to fix this
+    locale: locales[locale],
+  });
+
+  const author =
+    version.createdBy?.username || `${version.createdBy?.firstname} ${version.createdBy?.lastname}`;
+
+  return (
+    <Flex
+      direction="column"
+      alignItems="flex-start"
+      gap={3}
+      hasRadius
+      borderWidth="1px"
+      borderStyle="solid"
+      borderColor={isActive ? 'primary600' : 'neutral200'}
+      paddingTop={5}
+      paddingBottom={5}
+      paddingLeft={5}
+      paddingRight={5}
+      as={Link}
+      to={`?${stringify({ ...query, id: version.id })}`}
+      style={{ textDecoration: 'none' }}
+    >
+      <Flex direction="column" gap={1} alignItems="flex-start">
+        <Typography as="h3" fontWeight="semiBold">
+          {formatDate(version.createdAt, {
+            day: 'numeric',
+            month: 'numeric',
+            year: 'numeric',
+            hour: '2-digit',
+            minute: '2-digit',
+          })}
+        </Typography>
+        <Typography as="p" textColor="neutral600">
+          {formatMessage(
+            {
+              id: 'todo',
+              defaultMessage:
+                '{distanceToNow}{isAnonymous, select, true {} other { by {author}}}{isCurrent, select, true { <b>(current)</b>} other {}}',
+            },
+            {
+              distanceToNow,
+              author,
+              isAnonymous: version.createdBy ? 'false' : 'true',
+              isCurrent: isCurrent ? 'true' : 'false',
+              b: BlueChunks,
+            }
+          )}
+        </Typography>
+      </Flex>
+      {version.status && (
+        <Box
+          background={statusData.background}
+          borderStyle="solid"
+          borderWidth="1px"
+          borderColor={statusData.border}
+          hasRadius
+          paddingLeft="6px"
+          paddingRight="6px"
+          paddingTop="2px"
+          paddingBottom="2px"
+        >
+          <Typography variant="pi" fontWeight="bold" textColor={statusData.text}>
+            {formatMessage(statusData.message)}
+          </Typography>
+        </Box>
+      )}
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * VersionsList
+ * -----------------------------------------------------------------------------------------------*/
+
+interface VersionsListProps {
+  contentType: UID.ContentType;
+  documentId?: Entity.ID;
+}
+
+const VersionsList = ({ contentType, documentId }: VersionsListProps) => {
+  const { formatMessage } = useIntl();
+  const navigate = useNavigate();
+
+  // Parse state from query params
+  const [{ query }] = useQueryParams<{
+    page?: number;
+    id?: number;
+    plugins?: Record<string, unknown>;
+  }>();
+  const validQueryParams = buildValidGetParams(query);
+  const page = validQueryParams.page ? Number(validQueryParams.page) : 1;
+  const locale = validQueryParams.locale;
```

Comment:
I'd still just pass all the query params, that way if someone extends the controller they get their plugin params and can do w/e they want to 😄